Pokenav Data

garakmon opened the pull request.

PikalaxALT commented on src/data/text/trainers_eye_messages.h:
Define a macro for each of these:
```c
#define MCFLAVOR(name) {gMatchCallFlavorText_##name##_Strategy, ... }
```

PikalaxALT replied:
Index these with the trainer rematch enums:
```c
    [REMATCH_ROSE] = MCFLAVOR(AromaLadyRose),
```

PikalaxALT commented on src/pokenav_unk_3.c:
```diff
@@ -37,9 +37,10 @@ static u32 sub_81CACF8(struct Pokenav3Struct *);
+#include "data/text/trainers_eye_messages.h"
+
+const u8 gUnknown_08622508[] = {0, 2};
+const u8 gUnknown_0862250A[] = {0, 1, 2, 0, 0, 0};
```
These trailing 0s may be little more than padding (0862250D-->08622510). See if it compiles if you remove them.

PikalaxALT commented on src/pokenav_unk_2.c:
```diff
@@ -1,6 +1,261 @@
+    }
+};
+
+u32 sub_81C9C6C(int);
```
For consistency's sake, LoopedTask should be `u32 func(s32)`. Some newer versions of GCC complain if you declare s32 as int instead.

PikalaxALT commented on src/pokenav_unk_9.c:
```diff
@@ -0,0 +1,75 @@
+    .baseBlock = 20
+};
+
+// TODO: these are probably not strings?
```
They definitely look like strings. Color specs for gender symbols.

garakmon commented:
updated.
fixed everything but the gitattributes.
I would have thought the line `*.bin = binary` would prevent the diffs from appearing like that, so idk what the issue is or how to resolve it.

PikalaxALT commented on src/data/text/match_call_messages.h:
```diff
@@ -0,0 +1,476 @@
+#define MCFLAVOR(name) {gMatchCallFlavorText_##name##_Strategy, \
```
I can't read this macro name and not laugh

DizzyEggg replied:
lmao
